tensorboard/data/server/data_compat.rs

```diff
@@ -296,6 +314,12 @@ mod tests {
                     tensor_content: f64::to_le_bytes(0.125).to_vec(),
                     ..Default::default()
                 },
+                pb::TensorProto {
+                    dtype: pb::DataType::DtDouble.into(),
+                    tensor_shape: None, // no explicit tensor shape; treated as rank 0
+                    tensor_content: f64::to_le_bytes(0.125).to_vec(),
```

Comment:
Sure! I commend you on your apropos use of `try_into()`. I don’t
initially have a strong preference here. I guess getting rid of the
`size_of` check is nice and makes it clearer that this doesn’t panic?
e.g., you could typo `size_of::<f64>` in both branches. My tests would
catch this, but in your implementation the type system would catch it.

Okay, sold! and done (…with a deref coercion solely so that the code
fits on one line instead of being split onto five).

Reply:
Heh well I may have gleaned that part from the stackoverflow results for "rust convert slice to array halp" but I will pocket the commendation anyhow :) Yeah, I liked tying both the size check and the actual decoding together via the type system.